borg/cache.py

```diff
@@ -375,12 +378,12 @@ def add_chunk(self, id, data, stats, overwrite=False):
         data = self.key.encrypt(data)
         csize = len(data)
         self.repository.put(id, data, wait=False)
-        self.chunks[id] = (refcount + 1, size, csize)
+        self.chunks[id] = ChunkIndexEntry(refcount + 1, size, csize)
```

Comment:
Change chunks.add to use size/csize from the supplied values and not what's already in the hashindex?